### DISCUSSION TITLE
Study and communication

### DISCUSSION THREAD

**Discussion by ymkzpx** [duckdb/duckdb#3285] [General]
Hello, DuckDB developer. 

I am a graduate student in computer science. I have been reading the source code of DuckDB recently and am confused about the various data structures in DuckDB.

For example, SegmentBase, SegmentTree, ColumnSegment, DataChunk... .

I wonder how these data structures work together, and design.

I am looking forward to your reply.

Thank you very much.

**Comment by Alex-Monahan** [comment]
Hello and welcome!

I'm the documentation expert and a user of DuckDB, so I can't help explain the internals. However, could you tell us a bit more about what you are hoping to learn? What is your end goal of understanding DuckDB - are you hoping to help contribute in the future? Or is there a certain part of the system you are studying? 
Thanks!

**Comment by ymkzpx** [comment]
Hi Alex:
I focused on the storage tier and tried to place some of the data in Non-Volatile Memory.
